infrastructure/transform/main.tf

```diff
@@ -12,11 +12,66 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-resource "azurerm_databricks_workspace" "transform" {
+resource "azurerm_databricks_workspace" "databricks" {
   name                        = "${var.prefix}-${var.environment}-dbks"
   resource_group_name         = var.core_rg_name
   managed_resource_group_name = "${var.prefix}-${var.environment}-rg-dbks"
   location                    = var.core_rg_location
   sku                         = "standard"
   tags                        = var.tags
 }
+
+data "databricks_spark_version" "latest_lts" {
+  spark_version = var.spark_version
+
+  depends_on = [azurerm_databricks_workspace.databricks]
+}
+
+data "databricks_node_type" "smallest" {
+  # Providing no required configuration, Databricks will pick the smallest node possible
+  depends_on = [azurerm_databricks_workspace.databricks]
+}
+
+resource "databricks_cluster" "fixed_single_node" {
+  cluster_name            = "Fixed Job Cluster"
+  spark_version           = data.databricks_spark_version.latest_lts.id
+  node_type_id            = data.databricks_node_type.smallest.id
+  autotermination_minutes = 10
+
+  spark_conf = {
+    "spark.databricks.cluster.profile" : "singleNode"
+    "spark.master" : "local[*]"
+  }
+
+  custom_tags = {
+    "ResourceClass" = "SingleNode"
+  }
+
+  depends_on = [azurerm_databricks_workspace.databricks]
+}
+
+resource "azurerm_role_assignment" "adf_can_create_clusters" {
+  scope                = azurerm_databricks_workspace.databricks.id
+  role_definition_name = "Contributor"
+  principal_id         = azurerm_data_factory.adf.identity[0].principal_id
+}
+
+resource "azurerm_data_factory" "adf" {
+  name                = "${var.prefix}-${var.environment}-dbks"
```

Comment:
`adf` suffix might be more appropriate(?)

Reply:
Good spot! I got too excited with copy paste